---
title: "Posterior predictive check for bulk RNA sequencing data"
output: github_document
---


The input data set is a tidy representation of a differential gene transcript abundance analysis

```{r}
library(tidyverse)
library(ppcSeq)
```

```{r eval=FALSE}
counts # Available at ppcSeq::counts

```

You can convert a list of BAM/SAM files into a tidy data frame of annotated counts

```{r cache=TRUE, warning=FALSE, message=FALSE}
counts.ppc = 
	counts %>% 
		mutate(is_significant = FDR < 0.05) %>% 
		ppc_seq(
			formula = ~ Label + W, 
			significance_column = "PValue", 
			do_check_column = "is_significant", 
			value_column = "value"
		)
```

The new posterior predictive check has been added to the original data frame

```{r, cache=TRUE}
counts.ppc %>% 
	distinct(symbol, FDR, `ppc samples failed`, plot)

```

The new data frame contains plots for each gene

We can visualise the top five differentially transcribed genes

```{r, cache=TRUE}
counts.ppc %>% 
	distinct(symbol, FDR, `ppc samples failed`, plot) %>%
	slice(1:5) %>%
	pull(plot) 

```
